src/erlcloud_cloudfront.erl

```diff
@@ -0,0 +1,233 @@
+     {smooth_streaming, "SmoothStreaming", boolean},
+     {default_ttl, "DefaultTTL", integer},
+     {max_ttl, "MaxTTL", integer},
+     {min_ttl, "MinTTL", integer},
+     {path_pattern, "PathPattern", optional_text},
+     {target_origin_id, "TargetOriginId", text},
+     {viewer_protocol_policy, "ViewerProtocolPolicy", text},
+     {forwarded_query_string, "ForwardedValues/QueryString", boolean},
+     {forwarded_query_string_cache_keys, "ForwardedValues/QueryStringCacheKeys/Items/Name", list},
+     {forwarded_cookies_forward, "ForwardedValues/Cookies/Forward", boolean},
+     {forwarded_cookies_whitelisted_names, "ForwardedValues/Cookies/WhitelistedNames/Items/Name", list},
+     {forwarded_headers, "ForwardedValues/Headers/Items/Name", list},
+     {trusted_signers, "TrustedSigners/Items/AwsAccountNumber", list},
+     {trusted_signer_required, "TrustedSigners/Enabled", boolean}
+    ], Node).
+
+
+-spec extract_custom_error_response(Node :: list()) -> proplist().
+extract_custom_error_response(Node) ->
+    erlcloud_xml:decode(
+      [
+       {error_caching_min_ttl, "ErrorCachingMinTTL", integer},
+       {error_code, "ErrorCode", integer},
+       {response_code, "ResponseCode", text},
+       {response_page_path, "ResponsePagePath", text}
+      ], Node).
+
+
+-spec extract_origin(Node :: list()) -> proplist().
+extract_origin(Node) ->
+    erlcloud_xml:decode(
+      [
+       {origin_id, "Id", text},
+       {origin_path, "OriginPath", optional_text},
+       {domain_name, "DomainName", optional_text},
+       {custom_headers, "CustomHeaders/Items/OriginCustomHeader", {map, fun extract_custom_header/1}},
+       {http_port, "CustomOriginConfig/HTTPPort", optional_integer},
+       {https_port, "CustomOriginConfig/HTTPSPort", optional_integer},
+       {origin_protocol_policy, "CustomOriginConfig/OriginProtocolPolicy", optional_text},
+       {origin_ssl_protocols, "CustomOriginConfig/OriginSslProtocols/Items/SslProtocol", list},
+       {origin_access_identity, "S3OriginConfig/OriginAccessIdentity", optional_text}
+      ], Node).
+
+-spec extract_custom_header(Node :: list()) -> proplist().
+extract_custom_header(Node) ->
+    erlcloud_xml:decode(
+      [
+       {name, "HeaderName", text},
+       {value, "HeaderValue", text}
+      ], Node).
+
+-spec extract_viewer_certificate(Node :: list()) -> proplist().
+extract_viewer_certificate(Node) ->
+    erlcloud_xml:decode(
+      [
+       {acm_certificate_arn, "ACMCertificateArn", optional_text},
+       {certificate, "Certificate", optional_text},
+       {certificate_source, "CertificateSource", optional_text},
+       {cloudfront_default_certificate, "CloudFrontDefaultCertificate", optional_text},
+       {iam_certificate_id, "IAMCertificateId", optional_text},
+       {minimum_protocol_version, "MinimumProtocolVersion", optional_text},
+       {ssl_support_method, "SSLSupportMethod", optional_text}
+      ], Node).
+
+
+flush_undefined(L) ->
```

Comment:
i do not see a need for a fun named like this. use filter direct in below please.